core/src/replay_stage.rs

```diff
@@ -1049,13 +1050,17 @@ impl ReplayStage {
         let mut verify_total = Measure::start("verify_and_process_entries");
         let mut entry_state = entries.start_verify(last_entry);
 
+        if let EntryVerificationStatus::Failure = entry_state.status() {
+            return handle_block_error(BlockError::InvalidEntryHash);
+        }
+
         let mut replay_elapsed = Measure::start("replay_elapsed");
         let res =
             blocktree_processor::process_entries(bank, entries, true, transaction_status_sender);
         replay_elapsed.stop();
         bank_progress.stats.replay_elapsed += replay_elapsed.as_us();
 
-        if !entry_state.finish_verify(entries) {
+        if let EntryVerificationStatus::Failure = entry_state.finish_verify(entries) {
```

Comment:
`if EntryVerificationStatus::Failure == entry_state.finish_verify(entries) {`

would be clearer for these, imo.